packages/wrangler/src/pages/publish.tsx

```diff
@@ -256,18 +257,24 @@ export const Handler = async ({
 	const routesOutputPath = join(tmpdir(), `_routes-${Math.random()}.json`);
 	if (existsSync(functionsDirectory)) {
 		const outfile = join(tmpdir(), `./functionsWorker-${Math.random()}.js`);
-
-		await new Promise((resolve) =>
```

Comment:
Much nicer. No idea why we were doing it that old way. Is there _maybe_ a risk that even though `await buildFunctions` returns, it hasn't finished writing to disk yet? Surely not.

Reply:
I wondered about this, but looking inside `buildFunctions()`, everything is `await`'d anyway. We also don't use the `onEnd` thing in `wrangler pages functions build`, so I'm fairly sure it's safe.